metaflow/client/core.py

```diff
@@ -475,6 +475,44 @@ def __contains__(self, id: str):
         """
         return bool(self._get_child(id))
 
+    def __setstate__(self, state):
+        """
+        This function is used during the unpickling operation.
+        More info here https://docs.python.org/3/library/pickle.html#object.__setstate__
+        """
+        if isinstance(state, dict):
+            # For backward compatibility: handles pickled objects that were serialized as a `dict`
+            self.__init__(
+                pathspec=state.get("_pathspec", None),
+                attempt=state.get("_attempt", None),
+                _namespace_check=state.get("_namespace_check", None),
```

Comment:
this should default to `True` as this is what it defaults to in init and what objects had before we introduced this namespace_check thing.